dvc/repo/gc.py

```diff
@@ -41,15 +40,27 @@ def gc(
 
         used = NamedCache()
         for repo in all_repos + [self]:
+            exclude_revs = (
+                repo.scm.list_branches()
```

Comment:
Let's not spread scm logic all over codebase. `brancher` is not only used in gc, it is also used in push/pull and metrics. This PR meant to be very simple and local - we change how we accept things from user and translate it to internal calls, which are not changed.